keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,152 @@
+  - [Test Scenarios](#test-scenarios)
+  - [Tools](#tools)
+  - [Execution Environment](#execution-environment)
+  - [Future Enhancements](#future-enhancements)
+  - [Benefits](#benefits)
+- [Graduation Criteria](#graduation-criteria)
+  - [Beta -&gt; GA](#beta---ga)
+  - [GA -&gt; Future](#ga---future)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Summary
+
+We are proposing to add more testcases for network performance benchmarking.
+
+## Motivation
+
+As Kubernetes is increasingly being adopted by various telco cloud providers for deployments, 
+it brings additional networking requirements/scenarios to be supported.  
+With wide array of network functions being containerized, the need to analyse and 
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 2 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
+
+b) netperf (https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf) : This framework 
+   measures only the throughput between client-server pods for various MSS(Maximum Segment size). 
+   Other performance metrics such as latency , loss-rate , response time are not measured.
+
+### Goals
+
+The primary goal of this proposal is to enhance the testcases to cover various 
+networking scenarios :
+
+a)  IPv4 , IPv6 , Dual-Stack.
+b)  Protocols: TCP, UDP, HTTP/HTTPS.
+c)  With services such as ClusterIP , NodePort etc.
+
+We intend to submit tests to measure the following network performance metrics for various 
+scenarios such as different services (ClusterIP/NodePort), with IPv4/IPv6 configured etc.
+
+a)  Throughput (For TCP traffic)
+b)  Loss rate (For UDP traffic)
+c)  Latency/Jitter (For UDP/HTTP/HTTPS)
+d)  Response time (For HTTP/HTTPs)
+
+Our important goal is to make sure these benchmarking testcases are run periodically,
+so that any degradation in performance is detected in early phase of feature delivery.
+
+### Non-goals
+
+Benchmarking and measurements with intermediate services like ingress controllers, load-balancers 
+are out of scope of this proposal.
+
+### Challenges and Open Questions [optional]
+
+1) Which is the suitable hosting framework for proposed tests ? 
+2) Periodic execution of these proposed tests ?
+
+
+## Proposal
+
+### Test Scenarios
+
+As captured in the link https://docs.google.com/document/d/1_760ptAHBY_c56slWhjl6PJQOrf__vQFJXkL58ewg1M/edit
+we propose to have 2 broad categories based on how client PODs can communicate with server PODs.
+
+The testcase will cover different protocols such as TCP, UDP, HTTP/HTTPS with IPv4/IPv6 addresses configured.
+
+The testcases will cover different client-to-server pod number ratio (i.e 1:N , 1:1 , N:1)
```

Comment:
are we going to build a matrix, is N a variable? 
also, we should consider if the pods are in the same node or not, that will impact highly the results (pod to pod in the same node )

Reply:
Yes N is variable. We can define the possible values like 5/10/20 etc.
But we can limit it to few values , otherwise the testsuite runtime will increase.

Yes , forgot to capture this important point of client-server pod co-located on same node or remotely located.

Reply:
Just for taking it into account, you'll need this data also for asking for resources :)